File: C++/sum-of-square-numbers.cpp
// Time:  O(sqrt(c) * logc)
// Space: O(1)

class Solution {
public:
    bool judgeSquareSum(int c) {
        for (long long a = 0; a * a <= c; ++a) {
            auto b = static_cast<int>(sqrt(c - a * a));
            if (a * a + b * b == c) {
                return true;
            }
        }
        return false;
    }
};

File: Python/sum-of-square-numbers.py
# Time:  O(sqrt(c) * logc)
# Space: O(1)

import math


class Solution(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        for a in xrange(int(math.sqrt(c))+1):
            b = int(math.sqrt(c-a**2))
            if a**2 + b**2 == c:
                return True
        return False

